moviebuff.go

```diff
@@ -34,42 +33,44 @@ var (
 )
 
 // Moviebuff allows to access to information in moviebuff using resource ids.
+type Moviebuff interface {
+	GetMovie(id string) (*Movie, error)
+	GetPerson(id string) (*Person, error)
+	GetEntity(id string) (*Entity, error)
+	GetResources(resourceType ResourceType, limit, page int) (*Resources, error)
+}
+
+type Config struct {
+	HostURL     string
+	StaticToken string
+}
 
 // Before accessing any API it need to be initialized.
 // The Moviebuff is a service that offers information about movies, people, entities.
-type Moviebuff struct {
-	token string
-	l     logger
+type moviebuff struct {
+	Config
 }
 
-// Init initialize the Moviebuff.
-// token is mandatory. logger is optional however.
-// Return the same Moviebuff object back
-func (m *Moviebuff) Init(token string, l logger) *Moviebuff {
-	if l == nil {
-		l = log.New(new(devNull), "", 0)
+//New returns a QubeAccount interface with a nil logger.
+func New(config Config) Moviebuff {
```

Comment:
What if they send empty config?

Reply:
We get an error at consumer end like `Error while fetching movie information`